pkg/river/parser/interp_string.go

```diff
@@ -0,0 +1,282 @@
+package parser
```

Comment:
On re-review, it seems pretty unnecessary for the interpStringScanner to be a near-copy of the regular scanner, especially with the escape handling. 

This can likely be replaced with either

* Regex, if regex can properly handle extra `{}` characters inside the interpolated string
* A simpler scanner which is only looking for a non-escaped `${` start sequence, and scans everything inside of that sequence until a matching `}` (while keeping track of inner `{}` pairs). 

If regex is used, performance should be benchmarked to see if it's reasonable, especially for longer strings.